lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart

```diff
@@ -0,0 +1,247 @@
+    });
+    var actionsAssigned = cascadingAssignments.any((cascade) {
+      final lhs = cascade.leftHandSide;
+      return lhs is PropertyAccess && lhs.propertyName.name == 'actions';
+    });
+
+    if (!storeAssigned) {
+      storeAssigned = true;
+      final storeValue =
+          _getNameOfVarOrFieldInScopeWithType(node, fluxStoreType) ?? 'null';
+      yieldNewCascadeSection(node, '..store = $storeValue');
+    }
+
+    if (!actionsAssigned) {
+      actionsAssigned = true;
+      final actionsValue =
+          _getNameOfVarOrFieldInScopeWithType(node, fluxActionsType) ?? 'null';
+      yieldNewCascadeSection(node, '..actions = $actionsValue');
+    }
+  }
+
+  void yieldNewCascadeSection(CascadeExpression node, String newSection) {
+    final offset = node.target.end;
+    yieldPatch(newSection, offset, offset);
+  }
+
+  @override
+  Future<void> generatePatches() async {
+    _result = await context.getResolvedUnit();
+    if (_result == null) {
+      throw Exception(
+          'Could not get resolved result for "${context.relativePath}"');
+    }
+    _result!.unit.accept(this);
+  }
+}
+
+class InScopeVariable {
+  final String name;
+  final DartType? type;
+
+  InScopeVariable(this.name, this.type);
+}
+
+String? _getNameOfVarOrFieldInScopeWithType(AstNode node, DartType type) {
+  final mostInScopeVariables = node.ancestors.expand((ancestor) sync* {
+    if (ancestor is FunctionDeclaration) {
+      // Function arguments
+      final element = ancestor.declaredElement;
+      if (element != null) {
+        yield* element.parameters.map((p) => InScopeVariable(p.name, p.type));
+      }
+    } else if (ancestor is Block) {
+      // Variables declared in the block (function body, if/else block, etc.)
+      yield* ancestor.statements
+          .whereType<VariableDeclarationStatement>()
+          .expand((d) => d.variables.variables)
+          .map((v) => InScopeVariable(v.name.lexeme, v.declaredElement?.type));
+    } else if (ancestor is ClassDeclaration) {
+      // Class fields
+      final element = ancestor.declaredElement;
+      if (element != null) {
+        yield* element.fields.map((f) => InScopeVariable(f.name, f.type));
+      }
+    } else if (ancestor is CompilationUnit) {
+      // Top-level variables
+      yield* ancestor.declarations
+          .whereType<TopLevelVariableDeclaration>()
+          .expand((d) => d.variables.variables)
+          .map((v) => InScopeVariable(v.name.lexeme, v.declaredElement?.type));
+    }
+  });
+
+  // Usually we'd grab typeSystem from the ResolvedUnitResult, but we don't have access to that
+  // in this class, so just get it from the compilation unit.
+  final typeSystem =
+      (node.root as CompilationUnit).declaredElement!.library.typeSystem;
+  bool isMatchingType(DartType? maybeMatchingType) =>
+      maybeMatchingType != null &&
+      maybeMatchingType is! DynamicType &&
+      type is! DynamicType &&
+      typeSystem.isAssignableTo(maybeMatchingType, type);
```

Comment:
I noticed some places where store was getting assigned to some unexpected values (for example, [here](https://sourcegraph.plat.workiva.net/organizations/Workiva/batch-changes/apply/QmF0Y2hTcGVjOiJkYTRmZDc0OS0zYjNlLTRjMDEtYTU1My1lYzQxYjhmM2UyN2Yi?visible=33#diff-e8eeaa246874b37275c86779da0b0843)). 

It looks like for at least one (`SidePanelContents`), it's because the TStore is `Null`. Perhaps we should have another check for `type.isDartCoreNull`?

Come to think of it, we could probably just short-circuit in this method before we even start to search:
```dart
String? _getNameOfVarOrFieldInScopeWithType(AstNode node, DartType type) {
  if (type is DynamicType || type.isDartCoreNull) return null;
  // ...
```

Reply:
Fixed / added tests in https://github.com/Workiva/over_react_codemod/pull/268/commits/a603afe57c07b05115943f49fa5ad000708a2d84